### ISSUE TITLE
else after +1/-1 fails to parse

### ISSUE THREAD

**Issue by greghuc** [DanielXMoore/Civet#309]
### Coffeescript bug (parse error)

```
"civet coffeeCompat"

capped = (value) ->
  if value < 0
    -1
  else if value > 0
    1
  else
    value
    
console.log capped(-27) 
```
gives
```
"civet coffeeCompat"

capped = (value) ->
  if value < 0
    -1
  else if value > 0
      ^ ParseError
```

### Playground links

Coffeescript: https://coffeescript.org/#try:%0Acapped%20%3D%20(value)%20-%3E%0A%20%20if%20value%20%3C%200%0A%20%20%20%20-1%0A%20%20else%20if%20value%20%3E%200%0A%20%20%20%201%0A%20%20else%0A%20%20%20%20value%0A%20%20%20%20%0Aconsole.log%20capped(-27)
Civet: https://civet.dev/playground/?code=ImNpdmV0IGNvZmZlZUNvbXBhdCIKCmNhcHBlZCA9ICh2YWx1ZSkgLT4KICBpZiB2YWx1ZSA8IDAKICAgIC0xCiAgZWxzZSBpZiB2YWx1ZSA%2BIDAKICAgIDEKICBlbHNlCiAgICB2YWx1ZQogICAgCmNvbnNvbGUubG9nIGNhcHBlZCgtMjcpCg%3D%3D

**Comment by greghuc** [comment]
[Adding brackets round the if conditions](https://civet.dev/playground/?code=ImNpdmV0IGNvZmZlZUNvbXBhdCIKCmNhcHBlZCA9ICh2YWx1ZSkgLT4KICBpZiAodmFsdWUgPCAwKQogICAgLTEKICBlbHNlIGlmICh2YWx1ZSA%2BIDApCiAgICAxCiAgZWxzZQogICAgdmFsdWUKICAgIApjb25zb2xlLmxvZyBjYXBwZWQoLTI3KQoKCgoK) fixes the issue:

```
"civet coffeeCompat"

capped = (value) ->
  if (value < 0)
    -1
  else if (value > 0)
    1
  else
    value
    
console.log capped(-27)
```

**Comment by greghuc** [comment]
This bug is more of an FYI. I fixed it in the Coffeescript codebase I'm exploring porting to Civet by switching to the parsable form.

**Comment by edemaine** [comment]
This is a pretty curious bug.  The smallest reproduction I've found is [this](https://civet.dev/playground?code=aWYgeAogICt4CmVsc2UKICB4Cg%3D%3D):

```coffee
if x
  +x
else
  x
```

It seems crucial that there's a unary operation in the "then" clause, and then it's followed by an "else" clause.

**Comment by STRd6** [comment]
I think it has to do with the `__` in `BinaryOpRHS` inside of a `Condition` so it parses it like:

```
if (x
  +x)
else <- gives up
```

**Comment by STRd6** [comment]
Fixed in v0.5.61
